protomap/proto.go

```diff
@@ -359,3 +359,77 @@ func resolvedPath(basePath, annotatedPath *gpb.Path) *gpb.Path {
 	np.Elem = append(np.Elem, annotatedPath.Elem[len(basePath.Elem):]...)
 	return np
 }
+
+// ProtoFromPaths takes an input proto.Message and adds the values that are specified in the map vals to
+// it, using basePath as the prefix to any paths within the vals map. The message, p, is modified in place.
+// The map, vals, is keyed by the gNMI path to the field which is annotated in the ygot generated protobuf,
+// the complete path is taken to be basePath + the key found in the map.
+func ProtoFromPaths(p proto.Message, vals map[*gpb.Path]interface{}, basePath *gpb.Path) error {
+	m := p.ProtoReflect()
+
+	directCh := map[*gpb.Path]interface{}{}
+	for p, v := range vals {
+		// TODO(robjs): needs fixing for compressed schemas.
+		l := 0
+		if basePath != nil {
+			l = len(basePath.Elem)
+		}
+		if len(p.Elem) == l+1 {
+			directCh[p] = v
+		}
+	}
+
+	var rangeErr error
+	unpopRange{m}.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
+		annotatedPath, err := annotatedSchemaPath(fd)
+		if err != nil {
+			rangeErr = err
+			return false
+		}
+
+		for _, ap := range annotatedPath {
+			for chp, chv := range directCh {
+				if proto.Equal(ap, chp) {
+					if fd.Kind() == protoreflect.MessageKind {
+						v, isWrap, err := makeWrapper(m, fd, chv)
+						if err != nil {
+							rangeErr = err
+							return false
+						}
+						if !isWrap {
+							// TODO(robjs): recurse into the message if it wasn't a wrapper
+							// type.
+							rangeErr = errors.New("unimplemented: child messages")
+							return false
+						}
+						m.Set(fd, protoreflect.ValueOfMessage(v))
+					}
+				}
+			}
+		}
+		return true
+	})
+
+	if rangeErr != nil {
+		return rangeErr
+	}
+
+	return nil
+}
+
+// makeWrapper generates a new message for field fd of the proto message m with the value set to val.
+// The field fd must describe a field that has a message type. An error is returned if the wrong
+// type of payload is provided for the message. The second, boolean, return argument specifies whether
+// the message provided was a known wrapper type.
+func makeWrapper(msg protoreflect.Message, fd protoreflect.FieldDescriptor, val interface{}) (protoreflect.Message, bool, error) {
+	newV := msg.NewField(fd)
+	switch newV.Message().Interface().(type) {
+	case *wpb.StringValue:
+		nsv, ok := val.(string)
+		if !ok {
+			return nil, false, fmt.Errorf("got non-string value for string field, field: %s, value: %v", fd.FullName(), val)
+		}
+		return (&wpb.StringValue{Value: nsv}).ProtoReflect(), true, nil
+	}
+	return nil, false, nil
```

Comment:
Since all cases in the switch end in a return, suggest to put the last return in a default statement. This is one condition where a missing return won't be flagged by the compiler.

Reply:
Done - the static analysis/linter does a good job on this - but I think your suggestion makes sense.